clients/spark/core/src/main/scala/io/treeverse/gc/DataLister.scala

```diff
@@ -50,9 +50,14 @@ class ParallelDataLister extends DataLister with Serializable {
   override def listData(configMapper: ConfigMapper, path: Path): DataFrame = {
     import spark.implicits._
     val slices = listPath(configMapper, path)
-    val objectsPath = path.toString // udf require serializable string and not Path
+    var objectsPath = path.toString // udf require serializable string and not Path
     val objectsUDF = udf((sliceId: String) => {
-      val slicePath = new Path(objectsPath, sliceId)
+      if (!objectsPath.endsWith("/")) {
+        objectsPath += "/"
+      }
+      val p =
+        s"$objectsPath$sliceId" // WA for https://issues.apache.org/jira/browse/HDFS-14762
```

Comment:
Just making sure: sliceId is always relative, right?

Also, why not just add the strings here?